main.rb

```diff
@@ -0,0 +1,175 @@
+# rubocop:disable Style/CyclomaticComplexity
+require './book'
+require './person'
+require './teacher'
+require './student'
+require './rental'
+
+class Library
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_all_books
+    if @books.empty?
+      print 'No books in library'
+      return
+    end
+    @books.each do |book|
+      print "Title: #{book.title.capitalize}, Author: #{book.author.capitalize}\n"
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp.capitalize
+    print 'Author: '
+    author = gets.chomp.capitalize
+    book = Book.new(title: title, author: author)
+    @books << book
+    puts "Book created successfully\n"
+  end
+
+  def list_all_people
+    if @people.empty?
+      puts 'Your Library is empty'
+      return
+    end
+    @people.each do |person|
+      print "[#{person.class.name}] Name: #{person.name.capitalize}, ID: #{person.id}, Age: #{person.age}\n"
+    end
+  end
+
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+
+    option = gets.chomp
+
+    case option
+    when '1'
+      print 'Age: '
+      age = gets.chomp
+      print 'Name: '
+      name = gets.chomp
+      print 'Classroom:'
+      classroom = gets.chomp
+      print 'Has parent permission? [Y/N]: '
+      permission_resp = gets.chomp
+      parent_permission = permission_resp.downcase == 'y'
+
+      student = Student.new(age: age, classroom: classroom, name: name, parent_permission: parent_permission)
+      @people.push(student)
+
+      puts "Person created successfuly\n"
+    when '2'
+      print 'Age: '
+      age = gets.chomp
+      print 'Name: '
+      name = gets.chomp
+      print 'Specialization: '
+      specialization = gets.chomp
+
+      teacher = Teacher.new(age: age, specialization: specialization, name: name, parent_permission: parent_permission)
+      @people.push(teacher)
+
+      puts "Person created successfuly\n"
+
+    else
+      puts 'Please choose number 1 or 2'
+      nil
+    end
+  end
+
+  def create_rental
+    if @people.empty? && @books.empty?
+      puts 'Your Library is empty'
+      return
+    end
+    puts 'Select a book by number'
+    @books.each_with_index do |book, i|
+      print "#{i}) Title: #{book.title}, Author: #{book.author}\n"
+    end
+
+    book_index = gets.chomp.to_i
+    book = @books[book_index]
+
+    puts 'Select a person by number'
+    @people.each_with_index do |person, i|
+      print "#{i}) [#{person.class}] Name: #{person.name.capitalize}, ID: #{person.id}, Age: #{person.age}\n"
+    end
+
+    person_index = gets.chomp.to_i
+    person = @people[person_index]
+
+    print "\nDate: "
+
+    date = gets.chomp
+
+    rental = Rental.new(date: date, person: person, book: book)
+    @rentals << rental
+
+    puts "Rental created successfully\n"
```

Comment:
- [ ] When I tried to create a rental, I got an error. I'm sure this can be fixed.

![retal](https://user-images.githubusercontent.com/53124548/201457308-98c4d1b4-6ee7-4c46-8d0e-1ad987e63aa1.png)